### PR TITLE
Test Added for Retry Before Failover

### PR DESCRIPTION
**Pull request by emreyigit**
A case is added to test that client is try to reconnect before doing a failover.

### CODE REVIEW

File: src/Hazelcast.Net.Tests/Clustering/FailoverTests.cs [hunk 1]
```diff
@@ -522,6 +523,94 @@ public async Task TestClientCannotFailoverToDifferentPartitionCount()
...
+
+            Assert.AreNotEqual(RcClusterAlternative.Id, client.ClusterName);
+
+            //Lets start cluster A again, we expect that client will reconnect to clusterA
+            HConsole.WriteLine(this, $"START: Members of Cluster A :{RcClusterPrimary.Id}");
+            membersA = await StartMembersAsync(RcClusterPrimary, 1);
+
+            Assert.AreEqual(RcClusterPrimary.Id, client.ClusterName);
+
+            await AssertClusterA(map, RcClusterPrimary.Id);
```
Comment (srknzl) [inline src/Hazelcast.Net.Tests/Clustering/FailoverTests.cs:608]: The following is better I think. 

```suggestion
            await AssertClusterA(map, client.ClusterName);
```

**Review by srknzl — CHANGES_REQUESTED** [review]
Looks good except for one minor comment

File: src/Hazelcast.Net.Tests/Clustering/FailoverTests.cs (hunk 1 above)
Comment (emreyigit) [reply src/Hazelcast.Net.Tests/Clustering/FailoverTests.cs]: Yeap, thanks.

**Review by srknzl — APPROVED** [review]

**Review by zpqrtbnk — APPROVED** [review]
approved

**Pull request merged**